packages/salesforcedx-vscode-core/src/commands/forceConfigSet.ts

```diff
@@ -4,33 +4,66 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import {
-  Command,
-  SfdxCommandBuilder
-} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { Config } from '@salesforce/core';
+import { LibraryCommandletExecutor } from '@salesforce/salesforcedx-utils-vscode/out/src';
+import { Row, Table } from '@salesforce/salesforcedx-utils-vscode/out/src/output';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/src/types';
+import { channelService, OUTPUT_CHANNEL } from '../channels';
 import { nls } from '../messages';
+import { getRootWorkspacePath } from '../util/rootWorkspace';
 import {
   EmptyParametersGatherer,
   SfdxCommandlet,
-  SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from './util';
 
-export class ForceConfigSetExecutor extends SfdxCommandletExecutor<{}> {
+const CONFIG_SET_EXECUTOR = 'force_config_set_org_text';
+const CONFIG_NAME = 'defaultusername'; // todo: localize
+
+export class ForceConfigSetExecutor extends LibraryCommandletExecutor<{}> {
   private usernameOrAlias: string;
   protected showChannelOutput = false;
+  private responses: Row[] = [];
 
-  public constructor(usernameOrAlias: string) {
-    super();
+  constructor(
+    usernameOrAlias: string
+  ) {
+    super(nls.localize(CONFIG_SET_EXECUTOR), CONFIG_SET_EXECUTOR, OUTPUT_CHANNEL);
     this.usernameOrAlias = `${usernameOrAlias}`.split(',')[0];
   }
 
-  public build(data: {}): Command {
-    return new SfdxCommandBuilder()
-      .withDescription(nls.localize('force_config_set_org_text'))
-      .withArg('force:config:set')
-      .withArg(`defaultusername=${this.usernameOrAlias}`)
-      .build();
+  public async run(response: ContinueResponse<string>): Promise<boolean> {
+    // In order to correctly setup Config, the process directory needs to be set to the current workspace directory
+    const path = getRootWorkspacePath(); // Get current workspace path
+    process.chdir(path); // Set process directory
+
+    const config = await Config.create(Config.getDefaultOptions());
+
+    config.set(CONFIG_NAME, this.usernameOrAlias);
+    await config.write();
+    this.responses.push({name: CONFIG_NAME, val: this.usernameOrAlias, success: String(true) });
```

Comment:
I think I was just copying what CLI seemed to be doing, so I can definitely see why there's some confusion here. Another thing to consider though is that the `createTable` method takes in a list, so it might make sense to keep it.